src/modules/frequency/infrastructure/frequency.dto.mapper.ts

```diff
@@ -21,5 +41,27 @@ export class FrequencyDTOMapper {
             activityName: prismaClass.activity?.name ?? "N/A",
         },
         };
-    }
+    };
+    static toStudentGeneralAttendanceDTO(prismaClass: PrismaStudentGeneralFrequency): StudentGeneralAttendanceResponseDTO {
+        const studentRef =  {
+            studentId: prismaClass.studentid,
+            fullName: prismaClass.fullname,
+            generalAttendanceAllowed: prismaClass.generalattendanceallowed==="true"?true:false,
+            status: prismaClass.status,
+    };
+        return studentRef;
+    };
+    static toStudentClassAttendanceItemDTO(result: PrismaStudentClassAttendance): StudentClassAttendanceItemDTO {
+        const formattedAttendance = result.attendance !== null 
+        ? `${(result.attendance * 100).toFixed(2)}%` 
+        : '0.00%';
```

Comment:
So long as its numeric, I would not worry about formatting on the backend
```suggestion
        const formattedAttendance = result.attendance !== null 
        ? `${(result.attendance * 100).toFixed(2)}%` 
        : '0.00%';
```

Reply:
Vou alterar isso diretamente no codigo e não colocar no "batch" de sugestões.